### PR TITLE
Compute real filters for DICS

### PR DESCRIPTION
**Pull request by brovelli**
I added the option to compute real filters for DICS and set it to True. 
Please check whether it is appropriate to set it to True and carefully look at consistency across modules.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
@brovelli do you have a ref for this one, or is this from the same paper?

**Comment by Eric89GXL** [comment]
Oh the ref is in #3840. Can you add a ref to the docstring with a `[1]_` and `References` section like here:

https://github.com/mne-tools/mne-python/blob/master/mne/stats/regression.py#L165

**Comment by brovelli** [comment]
The first time I saw the use of real filters was in this paper:
Hipp JF, Engel AK, Siegel M. Oscillatory synchronization in large-scale cortical networks predicts perception. Neuron. 2011 Jan 27;69(2):387-96.
https://www.ncbi.nlm.nih.gov/pubmed/21262474
The details are given in the supplementary material. See #3840

File: mne/beamformer/_dics.py [hunk 1]
```diff
@@ -65,6 +68,10 @@ def _apply_dics(data, info, tmin, forward, noise_csd, data_csd, reg,
         _prepare_beamformer_input(info, forward, label, picks, pick_ori)
 
     Cm = data_csd.data
+    
+    # Take real part of Cm to compute real filters
+    if real_filter:
+      Cm = Cm.real
```
Comment (Eric89GXL) [inline mne/beamformer/_dics.py]: four space indentation instead of two

File: mne/beamformer/_dics.py [hunk 2]
```diff
@@ -378,6 +394,10 @@ def csd_shapes(x):
                         (i + 1, n_csds))
 
         Cm = data_csd.data
+        
+        # Take real part of Cm to compute real filters
+        if real_filter:
+          Cm = Cm.real
```
Comment (Eric89GXL) [inline mne/beamformer/_dics.py]: four spaces instead of two

**Comment by agramfort** [comment]
thx @brovelli 

we'll take over.

any volunteer?

@brovelli just a last request. What is for you a good test that this works as expected?

**Comment by brovelli** [comment]
eh eh good question. I found the bug because I was comparing the same results I obtained in Matlab using Fieldtrip. I guess the best way would be through simulations. Instead of testing your code on real data, you may envision producing a simulated dataset:
i) generate a small network (say n=4) of source-level data with oscillatory component + an evoked response. You can use a neural mass model from spm (I can show you how) to simualted LFPs in a network where you can set the connections between nodes (you know the structure of the network). 
ii) Place these source data on n patches of cortex with realistic anatomy 
iii) Project it sensor-space and produce a dataset (EEG, MEG)

If you want more biologically realistic stuff, a better option is to use The Virtual Brain (http://www.thevirtualbrain.org/). If you want, I can ask the people here in Marseille (Demian Battaglia, Viktor Jirsa) if they have this kind of dataset.

Maybe this kind of dataset already exists, but I am not aware of. This would be very usefull for other sorfwares as well (Fieldtrip, Brainstorm, SPM).

I'll keep thinking about that....

**Comment by brovelli** [comment]
eh eh good question. I found the bug because I was comparing the same 
results I obtained in Matlab using Fieldtrip. I guess the best way would 
be through simulations. Instead of testing your code on real data, you 
may envision producing a simulated dataset:
i) generate a small network (say n=4) of source-level data with 
oscillatory component + an evoked response. You can use a neural mass 
model from spm (I can show you how) to simualted LFPs in a network where 
you can set the connections between nodes (you know the structure of the 
network).
ii) Place these source data on n patches of cortex with realistic anatomy
iii) Project it sensor-space and produce a dataset (EEG, MEG)

If you want more biologically realistic stuff, a better option is to use 
The Virtual Brain (http://www.thevirtualbrain.org/). If you want, I can 
ask the people here in Marseille (Demian Battaglia, Viktor Jirsa) if 
they have this kind of dataset.

Maybe this kind of dataset already exists, but I am not aware of. This 
would be very usefull for other sorfwares as well (Fieldtrip, 
Brainstorm, SPM).

I'll keep thinking about that....

**Pull request closed without merging**